org.eclipse.ui.console/src/org/eclipse/ui/console/TextConsoleViewer.java

```diff
@@ -105,7 +107,7 @@ public void documentAboutToBeChanged(DocumentEvent event) {
 
 		@Override
 		public void documentChanged(DocumentEvent event) {
-			updateLinks(event.fOffset);
+			updateLinks(mouseOffset);
```

Comment:
I don't think this is correct, if document is changed, previous mouse offset is most likely wrong, so the change offset as it was before was OK.

Reply:
The change offset as it was before was causing the bug.
This was causing the hyperlink which was being hovered before the document change
to be exited (unavailable until you move the mouse again).

Please see the conversations here:
https://github.com/eclipse-platform/eclipse.platform.debug/issues/141
and here:
https://github.com/eclipse-platform/eclipse.platform.debug/pull/142

The updateLinks method takes care of the linkEntered/linkExited logic.
This makes sense to be done only for the link which is being hovered (at the latest mouseOffset).

That's the only link which makes sense to be entered/exited after a document change.
Any other link will be entered/exited at mouseMove.